### ISSUE TITLE
docking.widgets.tree.internal.InProgressGTreeNode cannot be cast to class ghidra.app.plugin.core.symboltree.nodes.SymbolTreeNode 

### ISSUE THREAD

**Issue by MaKiPL** [NationalSecurityAgency/ghidra#96]
**Describe the bug**
Ghidra throws an exception first time user expands function list after running software. Issue is replicable. 

**To Reproduce**
1. Open analyzed project in CodeBrowser
2. Click '+' to expand functions list
3. Exception will now appear
4. Issue is not replicable anymore till you close and re-open project/software

**Expected behavior**
Exception is not thrown. 

**Environment (please complete the following information):**
 - OS: Windows 10

**Additional context**
```
Uncaught Exception! 
ClassCastException - class docking.widgets.tree.internal.InProgressGTreeNode cannot be cast to class ghidra.app.plugin.core.symboltree.nodes.SymbolTreeNode (docking.widgets.tree.internal.InProgressGTreeNode and ghidra.app.plugin.core.symboltree.nodes.SymbolTreeNode are in unnamed module of loader ghidra.GhidraClassLoader @7d4793a8)
java.lang.ClassCastException: class docking.widgets.tree.internal.InProgressGTreeNode cannot be cast to class ghidra.app.plugin.core.symboltree.nodes.SymbolTreeNode (docking.widgets.tree.internal.InProgressGTreeNode and ghidra.app.plugin.core.symboltree.nodes.SymbolTreeNode are in unnamed module of loader ghidra.GhidraClassLoader @7d4793a8)
	at ghidra.app.plugin.core.symboltree.actions.CutAction.isEnabledForContext(CutAction.java:72)
	at ghidra.app.plugin.core.symboltree.actions.SymbolTreeContextAction.isEnabledForContext(SymbolTreeContextAction.java:41)
	at docking.ComponentPlaceholder.contextChanged(ComponentPlaceholder.java:472)
	at docking.DockingWindowManager.contextChanged(DockingWindowManager.java:2072)
	at ghidra.framework.plugintool.PluginTool.contextChanged(PluginTool.java:1656)
	at docking.ComponentProvider.contextChanged(ComponentProvider.java:333)
	at ghidra.app.plugin.core.symboltree.SymbolTreeProvider.lambda$createTree$1(SymbolTreeProvider.java:154)
	at docking.widgets.tree.internal.GTreeSelectionModel.fireValueChanged(GTreeSelectionModel.java:118)
	at docking.widgets.tree.internal.GTreeSelectionModel.fireValueChanged(GTreeSelectionModel.java:75)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.notifyPathChange(DefaultTreeSelectionModel.java:1111)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.setSelectionPaths(DefaultTreeSelectionModel.java:297)
	at java.desktop/javax.swing.JTree.setSelectionInterval(JTree.java:2584)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Actions.increment(BasicTreeUI.java:4767)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Actions.actionPerformed(BasicTreeUI.java:4448)
	at java.desktop/javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1810)
	at java.desktop/javax.swing.JComponent.processKeyBinding(JComponent.java:2900)
	at java.desktop/javax.swing.JComponent.processKeyBindings(JComponent.java:2948)
	at java.desktop/javax.swing.JComponent.processKeyEvent(JComponent.java:2862)
	at java.desktop/java.awt.Component.processEvent(Component.java:6409)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5008)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4840)
	at java.desktop/java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1950)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:871)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1140)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:1010)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:836)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2772)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4840)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

---------------------------------------------------
Build Date: 2019-Feb-28 1236 EST
Ghidra Version: 9.0
Java Home: D:\ghidra_9.0\jdk-11.0.2
JVM Version: Oracle Corporation 11.0.2
OS: Windows 10 10.0 amd64
Workstation: DESKTOP-POLI9UN
```

**Comment by ryanmkurtz** [comment]
I cannot reproduce this.  Are you still having the issue?

**Comment by MaKiPL** [comment]
Looks like I also can't reproduce this issue on latest release- closing

**Comment by MaKiPL** [comment]
I was able to reproduce it with 100% success. Normally, when autoanalyze function is working, by expanding the "Functions" tab in Symbol tree there would be "Loading..." entry. By clicking fast enough this error would show. Naturally you can do it almost every time by clicking on "Functions" and hitting right arrow twice. That would allow you to try to select the "Loading..." entry in Functions list. This is minor, non breaking issue, but I decided to reopen this for 100% stability.

**Comment by dragonmacher** [comment]
Thanks for the update.  This should be an easy fix now that we know what caused it.

**Comment by ryanmkurtz** [comment]
This will be fixed for 9.1.
